Packages/com.unity.inputsystem/InputSystem/Editor/AssetImporter/AssetContext.cs

```diff
@@ -0,0 +1,106 @@
+#if UNITY_EDITOR
+
+using System;
+using UnityEditor;
+using UnityEditor.AssetImporters;
+
+namespace UnityEngine.InputSystem.Editor
+{
+    /// <summary>
+    /// Represents an asset import or creation context.
+    /// </summary>
+    internal interface IAssetContext
+    {
+        /// <summary>
+        /// The associated asset destination path of the main object.
+        /// </summary>
+        string assetPath { get; }
+
+        /// <summary>
+        /// The associated source path.
+        /// </summary>
+        string sourcePath { get; }
+
+        /// <summary>
+        /// Adds an object to the asset.
+        /// </summary>
+        /// <param name="identifier">Object identifier</param>
+        /// <param name="obj">The object to be added.</param>
+        /// <param name="icon">Optional icon</param>
+        void AddObjectToAsset(string identifier, Object obj, Texture2D icon = null);
+
+        /// <summary>
+        /// Sets the main object of the asset.
+        /// </summary>
+        /// <param name="obj">The main object of the asset</param>
+        void SetMainObject(Object obj);
+
+        /// <summary>
+        /// Logs an error associated with the importing or creation of the asset.
+        /// </summary>
+        /// <param name="message">An error message. Never null.</param>
+        void LogError(string message);
+    }
+
+    struct AssetImporterAssetContext : IAssetContext
+    {
+        private readonly AssetImportContext m_Context;
+
+        public AssetImporterAssetContext(AssetImportContext context)
+        {
+            m_Context = context;
+        }
+
+        public string assetPath => m_Context.assetPath;
+        public string sourcePath => m_Context.assetPath;
+
+        public void AddObjectToAsset(string identifier, Object subAsset, Texture2D icon)
+        {
+            m_Context.AddObjectToAsset(identifier, subAsset, icon);
+        }
+
+        public void SetMainObject(Object obj)
+        {
+            // Note that importer doesn't need to do any storage
+            m_Context.SetMainObject(obj);
+        }
+
+        public void LogError(string message)
+        {
+            m_Context.LogImportError(message);
+        }
+    }
+
+    struct AssetDatabaseAssetContext : IAssetContext
```

Comment:
I don't see this being used anywhere - why the extra complexity here?

Reply:
I will likely undo this part as mentioned in a previous comment about this work, it was helpful during the transition phase to remove code bloat.

Reply:
This has now been removed. Will mark this resolved.